File: 01gpio/04pwm/pwm_wpi_hw.js
var wpi = require('wiring-pi');
var pin = 18;

wpi.setup('gpio');
wpi.pinMode(pin, wpi.PWM_OUTPUT);
while(true){
	for(var i=0; i<=1024; i++){
		wpi.pwmWrite(pin, i);
		wpi.delay(1);
	}
	for(var i=1024; i>=0; i--){
		wpi.pwmWrite(pin, i);
		wpi.delay(1);
	}
			
}

File: 01gpio/04pwm/pwm_wpi_hw.py
# Notice!! Run with sudo
import wiringpi as wpi
PIN_LED = 18

wpi.wiringPiSetupGpio()
wpi.pinMode(PIN_LED, wpi.PWM_OUTPUT)

while True:
	for i in range(0, 1024):
		wpi.pwmWrite(PIN_LED, i)
		wpi.delay(1)
	for i in range(1024, -1, -1):
		wpi.pwmWrite(PIN_LED, i)
		wpi.delay(1)
